frame/bribedao/src/lib.rs

```diff
@@ -0,0 +1,315 @@
+
+			let id = <Self as Bribe>::create_bribe(request.clone())?;
+			Self::deposit_event(Event::BribeCreated { id, request });
+			Ok(().into())
+		}
+
+		/// Sell Votes request
+		#[pallet::weight(10_000)]
+		pub fn take_bribe(
+			origin: OriginFor<T>,
+			request: TakeBribeRequest<T>,
+		) -> DispatchResultWithPostInfo {
+			let _from = ensure_signed(origin)?;
+			let bribe_index = request.bribe_index;
+			let bribe_taken = <Self as Bribe>::take_bribe(request.clone())?;
+			if bribe_taken {
+				Self::deposit_event(Event::BribeTaken { id: bribe_index, request });
+			}
+			Ok(().into())
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		InvalidBribe,
+		InvalidIndex,
+		NotEnoughFunds,
+		NotEnoughStake,
+		PriceNotRequested,
+		AlreadyBribed,
+		EmptySupply,
+		CantFreezeFunds,
+		ReleaseFailed,
+		BribeDeletionFailed,
+		InvalidVote,
+	}
+
+	// offchain indexing
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn offchain_worker(_b: T::BlockNumber) {
+			log::info!("Indexing request offchain");
+		}
+	}
+
+	impl<T: Config> Bribe for Pallet<T> {
+		type BribeIndex = BribeIndex;
+		type AccountId = T::AccountId;
+		type ReferendumIndex = ReferendumIndex;
+		type Balance = T::Balance;
+		type Conviction = T::Conviction;
+		type CurrencyId = T::CurrencyId;
+
+		/// Register new bribe request
+		fn create_bribe(request: CreateBribeRequest<T>) -> Result<Self::BribeIndex, DispatchError> {
+			Self::do_create_bribe(request)
+		}
+
+		/// Register the votes a user wants to sell
+		fn take_bribe(request: TakeBribeRequest<T>) -> Result<bool, DispatchError> {
+			Self::do_take_bribe(request)
+		}
+
+		/// Delete a finished Bribe Request
+		fn delete_bribe(request: DeleteBribeRequest) -> Result<bool, DispatchError> {
+			// todo, make sure this function can not be abused
+			Self::do_delete_bribe(request)
+			//			Ok(true)
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Create a new Bribe Request
+		fn do_create_bribe(request: CreateBribeRequest<T>) -> Result<BribeIndex, DispatchError> {
+			let id = BribeCount::<T>::mutate(|id| {
+				*id += 1;
+				*id
+			});
+
+			ensure!(!BribeRequests::<T>::contains_key(id), Error::<T>::AlreadyBribed); //dont duplicate briberequest if we already have it
+
+			BribeRequests::<T>::insert(id, request);
+			Ok(id)
+		}
+
+		/// Find votes for a bribe request
+		fn do_match_votes(bribe_index: BribeIndex) -> Result<bool, DispatchError> {
+			ensure!(BribeRequests::<T>::contains_key(bribe_index), Error::<T>::InvalidIndex);
+
+			let bribe_request = BribeRequests::<T>::get(bribe_index).unwrap();
+
+			let ref_index = bribe_request.ref_index;
+			// Yield all the bribe votes for sale with the same ref index
+
+			let loot: Vec<BribesStorage> = Fastvec::<T>::get().find_all_pid(ref_index);
+
+			if !loot.is_empty() {
+				let mut spendamount: u32 = 0;
+
+				for bribes in loot {
+					// Cast Vote
+
+					let vote = Vote { aye: bribe_request.is_aye, conviction: Default::default() }; //todo
```

Comment:
What is conviction used for?

Reply:
https://crates.parity.io/pallet_democracy/enum.Conviction.html, Locks in votes for X amount of time, Changed to None since the pallet will try to match the votes with a buyer and vote as soon as possible